### PR TITLE
added language selector

### PR DESCRIPTION
**Pull request by luwol03**
| Status  | Type  |
| :---: | :---: |
| :white_check_mark: ready | Feature |

## Description

add language selector to settings page

## TODO

- [x] improve language selector component
- [x] add flags to selector
- [x] add language selector to selection screen

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: src/Components/Form/Select/Select.jsx [hunk 1]
```diff
@@ -4,24 +4,26 @@ import ReactSelect, { components } from "react-select";
 
 import "./Select.scss";
 
-import useCountryFlag from "../../../hooks/userCountryFlag";
+import useCountryFlag from "../../../hooks/useCountryFlag";
```
Comment (luwol03) [inline src/Components/Form/Select/Select.jsx:7]: add .js

File: src/Components/LanguageSelector/LanguageSelector.jsx [hunk 2]
```diff
@@ -21,13 +21,17 @@ const LanguageSelector = () => {
...
             ({ code, name }) => ({
               value: code,
-              label: name,
+              label: (
+                <SelectOptionWithFlag name={name} foreignLanguageCode={code} />
+              ),
             })
           )[0]
         }
         options={languages.map(({ code, name }) => ({
```
Comment (luwol03) [inline src/Components/LanguageSelector/LanguageSelector.jsx:29]: maybe we can move this map function to a named function and reuse it.

File: src/Components/Form/Select/Select.jsx (hunk 1 above)
Comment (zikowang) [reply src/Components/Form/Select/Select.jsx]: done

File: src/Components/LanguageSelector/LanguageSelector.jsx (hunk 2 above)
Comment (zikowang) [reply src/Components/LanguageSelector/LanguageSelector.jsx:29]: I am not sure if i understand you. Could you give an example for what you mean?

**Pull request merged**